src/Glpi/Form/Destination/CommonITILField/EntityFieldStrategy.php

```diff
@@ -0,0 +1,162 @@
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Form\Destination\CommonITILField;
+
+use Entity;
+use Glpi\Form\AnswersSet;
+use Glpi\Form\QuestionType\QuestionTypeItem;
+use Glpi\Form\QuestionType\QuestionTypeRequester;
+use Group;
+use Group_User;
+use Session;
+use User;
+
+enum EntityFieldStrategy: string
+{
+    case FROM_FORM            = 'from_form';
+    case FROM_FIRST_REQUESTER = 'from_first_requester';
+    case FROM_USER            = 'from_user';
+    case SPECIFIC_VALUE       = 'specific_value';
+    case SPECIFIC_ANSWER      = 'specific_answer';
+    case LAST_VALID_ANSWER    = 'last_valid_answer';
+
+    public function getLabel(): string
+    {
+        return match ($this) {
+            self::FROM_FORM            => __("From form"),
+            self::FROM_FIRST_REQUESTER => __("From first requester"),
+            self::FROM_USER            => __("From user"),
+            self::SPECIFIC_VALUE       => __("Specific entity"),
+            self::SPECIFIC_ANSWER      => __("Answer from a specific question"),
+            self::LAST_VALID_ANSWER    => __('Answer to last "Entity" item question'),
+        };
+    }
+
+    public function getEntityID(
+        EntityFieldConfig $config,
+        AnswersSet $answers_set,
+    ): ?int {
+        return match ($this) {
+            self::FROM_FORM            => $answers_set->getItem()->fields['entities_id'],
+            self::FROM_FIRST_REQUESTER => $this->getEntityIDForFirstRequester($answers_set),
+            self::FROM_USER            => Session::getActiveEntity(),
+            self::SPECIFIC_VALUE       => $config->getSpecificEntityId(),
+            self::SPECIFIC_ANSWER      => $this->getEntityIDForSpecificAnswer(
+                $config->getSpecificQuestionId(),
+                $answers_set
+            ),
+            self::LAST_VALID_ANSWER => $this->getEntityIDForLastValidAnswer($answers_set),
+        };
+    }
+
+    private function getEntityIDForFirstRequester(
```

Comment:
IMHO, This should be more complex than this.

First, you must be sure that all requesters can access a common entity. When creating a ticket in GLPI, everytime you add/remove a requester, the entities list is filtered using `CommonITILObject::getEntitiesForRequesters()`. You should never create a ticket that cannot be accessed by one of its actor.

Second, the `entities_id` field of both `User` and `Group` items was not initially made to be used as a "default entity to use when creating items". IMHO, a new `entities_id_for_creation` field should be introduced here. We already explored this with @orthagh some months ago. It could be usefull to specify what entity to use when creating items from the API, the CalDAV, ... Anyway, it should probably be done in another PR.

I propose to remove the `FROM_FIRST_REQUESTER` and `FROM_USER` options for the moment, on handle them in another PR.